driver/task_test.go

```diff
@@ -244,17 +244,17 @@ func TestTask_ProviderIDs(t *testing.T) {
 
 	cases := []struct {
 		name     string
-		task     Task
+		task     *Task
```

Comment:
`Task` contains a mutex. When we loop through cases later in the test, the loop creates copies of the cases, including copying the `Task` object and its mutex. We should not copy locks (e.g. mutex) by value.